### PR TITLE
H-199: Add Linear connection ontology types

### PR DESCRIPTION
**Pull request by benwerner01**
## 🌟 What is the purpose of this PR?

This PR adds an `@linear` HASH user, with the `User`, `Organization` and `Issue` ontology types.

## 🔗 Related links

- ...

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## ❓ How to test this?

1. Set the `LINEAR_CLIENT_ID` environment variable
2. Run the local developer environment, which should seed the types in the database

### CODE REVIEW

File: apps/hash-api/src/graph/linear-user.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { Logger } from "@local/hash-backend-utils/logger";
+
+import { createKratosIdentity } from "../auth/ory-kratos";
+import { ImpureGraphContext } from "./index";
+import { createUser, getUserByShortname } from "./knowledge/system-types/user";
+import { systemUserAccountId } from "./system-user";
+
+const linearUserShortname = "linear";
```
[the remark this answers is not in the document]
Comment (benwerner01) [reply apps/hash-api/src/graph/linear-user.ts]: /ignore reason This username will always be `linear` so can be hardcoded

**Review by TimDiekmann — DISMISSED** [review]

File: apps/hash-api/src/graph/linear-types.ts [hunk 2]
```diff
@@ -0,0 +1,1436 @@
...
+      {
+        propertyType: identifierPropertyType,
+        required: true,
+      },
+      {
+        propertyType: numberPropertyType,
+        required: true,
+      },
+      {
+        propertyType: previousIdentifierPropertyType,
```
Comment (TimDiekmann) [inline apps/hash-api/src/graph/linear-types.ts:1166]:
```suggestion
        propertyType: previousIdentifiersPropertyType,
```

Comment (benwerner01) [reply apps/hash-api/src/graph/linear-types.ts:1166]: This is intentional so that the property type is reusable on entity types whether in an array or not

Comment (TimDiekmann) [reply apps/hash-api/src/graph/linear-types.ts:1166]: Discussed offline, the property types should be singular

**Review by TimDiekmann — APPROVED** [review]

**Pull request merged**